### PR TITLE
Graph multi series select

### PR DESCRIPTION
**Pull request by boyskila**
In this PR is implemented multiple series selection from the Graph legend. The behavior is as follows.
   1. Clicking on an item will make it active and all other items will be deselected
   2. Clicking on an item will disable hover effect on the Graph series
   3. Clicking on an item + holding the CTRL key will not deselect other items if there are such
   4. Once multiple selections are made just clicking on an active item(without Ctrl key) will make it 
       inactive but all other selections will remain
Fixes: #6268

### CODE REVIEW

**Comment by boyskila** [comment]
@juliusv @roidelapluie please review.

File: web/ui/react-app/src/graph/Graph.tsx [hunk 1]
```diff
@@ -123,6 +151,11 @@ class Graph extends PureComponent<GraphProps, GraphState> {
               <SeriesName labels={labels} format />
             </div>
           ))}
+          {chartData.length > 1 && (
+            <div className="pl-1 mt-1 text-muted" style={{ fontSize: 13 }}>
+              Click: toogle serie, CTRL + click: toogle multiple series
```
Comment (juliusv) [inline web/ui/react-app/src/graph/Graph.tsx]: Can we style this identically to the query stats?

"toogle" -> "toggle"
"serie" -> "series"

**Comment by juliusv** [comment]
@boyskila Tests are still failing, they probably need an update?

**Comment by juliusv** [comment]
Looks great besides the nits!

**Comment by roidelapluie** [comment]
Excellent. I have tested it and have 2 comments:

A) Legend
Click: toggle series, CTRL + click: toggle multiple series
Maybe: Click: *select* series, CTRL + click: toggle multiple series

B) CTRL+Click behaviour when no one is selected
When no timeserie is selected, I think that CTRL+CLICK should "remove" that timeserie (e.g they all become selected but that one).

Good job!

**Comment by boyskila** [comment]
> Excellent. I have tested it and have 2 comments:
> 
> A) Legend
> Click: toggle series, CTRL + click: toggle multiple series
> Maybe: Click: _select_ series, CTRL + click: toggle multiple series
> 
> B) CTRL+Click behaviour when no one is selected
> When no timeserie is selected, I think that CTRL+CLICK should "remove" that timeserie (e.g they all become selected but that one).
> 
> Good job!

Thanks for the feedback @roidelapluie @juliusv  ! I think ```Click: select series``` is more appropriate.
Regarding point B If I understand correctly you suggest on ```Ctrl + click``` to deselect the series but if there is series already selected to continue working as described in point A.
This functionality eventually can be delegated to ```Shift + click``` wdyt?

**Comment by roidelapluie** [comment]
No

B is:

From :

I "Click" federate_1 and I have:




From:

I "CTRL+Click" federate_1 and I have:

**Comment by boyskila** [comment]
Cool! So I understand it correctly. This is what I describe in the comment above :). Just wonder isn't better Shift + click to handle it and to leave Ctrl + click just to toggle multiple series.

**Comment by roidelapluie** [comment]
I prefer ctrl because it is the normal ctrl behaviour, we just need to consider than they are all selected at the beginning

**Comment by roidelapluie** [comment]
ctrl+click on a selected TS deselects it unless they are all selected ; that is a bug

**Comment by boyskila** [comment]
Yeah, I'm Ok with that. @juliusv do you have any comment on that or to proceed?

**Comment by juliusv** [comment]
I trust you two :)

**Comment by boyskila** [comment]
@juliusv @roidelapluie ctrl + click adjustments are done. Please review when you have time, thanks.
P.s did small refactoring as well by extracting the Legend as a component.

**Comment by roidelapluie** [comment]
Great work!

One last issue however:


When I click (without CTRL) the serie gets deselected instead of selected:

**Comment by boyskila** [comment]
> Great work!
> 
> One last issue however:
> 
> 
> When I click (without CTRL) the serie gets deselected instead of selected:
> 

Done

**Comment by juliusv** [comment]
Hmm, it doesn't seem to work as expected currently. E.g. if I graph `rate(node_cpu_seconds_total[5m])` on https://deploy-preview-6391--prometheus-react.netlify.com/ and I click on one of the series in the legend, it initially selects only the one series, clicking on it a second time also shows other series in the graph again, but in light color even after leaving the legend, and a third time does nothing anymore.

**Comment by juliusv** [comment]
Also, when reloading the graph (even with totally different series), it seems to remember the selected series (it should forget it).

**Comment by boyskila** [comment]
@juliusv Yes, thanks for catching those bugs. They appeared after the small refactoring I have made. I think now everything is working as expected.

File: web/ui/react-app/src/graph/Legend.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    const { selectedIndexes } = this.state;
+
+    let selected = [index];
+    if (ev.ctrlKey) {
+      const { chartData } = this.props;
+      if (selectedIndexes.includes(index)) {
+        selected = selectedIndexes.filter(idx => idx !== index);
+      } else {
+        selected =
+          // Flip the logic - In case non is selected ctrl + click should deselect clicked series.
```
Comment (juliusv) [inline web/ui/react-app/src/graph/Legend.tsx]: "non" -> "none"

File: web/ui/react-app/src/graph/Legend.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            : [...selectedIndexes, index]; // Select multiple.
+      }
+    } else if (selectedIndexes.length === 1 && selectedIndexes.includes(index)) {
+      selected = [];
+    }
+
+    this.setState({ selectedIndexes: selected });
+    this.props.onSeriesToggle(selected, index);
+  };
+  render() {
```
Comment (juliusv) [inline web/ui/react-app/src/graph/Legend.tsx]: nit: add empty line above

**Comment by juliusv** [comment]
Seems to behave pretty awesome now, just some nits!

**Pull request merged**